python/ELASTICACHE_REDIS_CLUSTER_AUTO_BACKUP_CHECK/ELASTICACHE_REDIS_CLUSTER_AUTO_BACKUP_CHECK.py

```diff
@@ -0,0 +1,443 @@
+import sys
+import datetime
+import boto3
+import botocore
+
+try:
+    import liblogging
+except ImportError:
+    pass
+
+##############
+# Parameters #
+##############
+
+# Define the default resource to report to Config Rules
+DEFAULT_RESOURCE_TYPE = 'AWS::ElastiCache::CacheCluster'
+DEFAULT_PARAMETER_VALUE = {'SnapshotRetentionPeriod': 15}
+# Set to True to get the lambda to assume the Role attached on the Config Service (useful for cross-account).
+ASSUME_ROLE_MODE = False
+
+# Other parameters (no change needed)
+CONFIG_ROLE_TIMEOUT_SECONDS = 900
+
+#############
+# Main Code #
+#############
+
+
+def get_replication_groups(ec_client):
+    replication_groups = []
+    marker = None
+    replication_groups_result = {}
+    while True:
+        if not marker:
+            replication_groups_result = ec_client.describe_replication_groups(MaxRecords=100)
+        else:
+            replication_groups_result = ec_client.describe_replication_groups(Marker=marker, MaxRecords=100)
+        replication_groups.extend(replication_groups_result['ReplicationGroups'])
+        if 'Marker' in replication_groups_result:
+            marker = replication_groups_result['Marker']
+        else:
+            return replication_groups
+
+
+def get_cache_clusters(ec_client):
+    cache_clusters = []
+    marker = None
+    cache_clusters_result = {}
+    while True:
+        if not marker:
+            cache_clusters_result = ec_client.describe_cache_clusters(MaxRecords=100, ShowCacheNodeInfo=False, ShowCacheClustersNotInReplicationGroups=True)
+        else:
+            cache_clusters_result = ec_client.describe_cache_clusters(Marker=marker, MaxRecords=100, ShowCacheNodeInfo=False, ShowCacheClustersNotInReplicationGroups=True)
+        cache_clusters.extend(cache_clusters_result['CacheClusters'])
+        if 'Marker' in cache_clusters_result:
+            marker = cache_clusters_result['Marker']
+        else:
+            return cache_clusters
+
+
+def generate_evaluations(cache_clusters, replication_groups, snapshot_retention_period, event):
+    evaluations = []
+    if cache_clusters:
+        for cluster in cache_clusters:
+            if cluster['Engine'] == 'redis':
+                if cluster['SnapshotRetentionLimit'] == 0:
+                    evaluations.append(build_evaluation(cluster['CacheClusterId'], 'NON_COMPLIANT', event, annotation="Automatic backup not enabled for Amazon ElastiCache cluster: {}".format(cluster['CacheClusterId'])))
+                elif cluster['SnapshotRetentionLimit'] < snapshot_retention_period:
+                    evaluations.append(build_evaluation(cluster['CacheClusterId'], 'NON_COMPLIANT', event, annotation='Automatic backup retention period for Amazon ElastiCache cluster {} is less then {} day(s).'.format(cluster['CacheClusterId'], snapshot_retention_period)))
+                else:
+                    evaluations.append(build_evaluation(cluster['CacheClusterId'], 'COMPLIANT', event))
+    if replication_groups:

```

Comment:
This is `if` as no value. The `for` will not be executed if the `replication_groups` is empty.